tasks/install_pe.sh

```diff
@@ -1,105 +1,39 @@
-#!/usr/bin/env bash
-# shellcheck disable=SC2154
-# shellcheck disable=SC2034
+#!/bin/bash
+# shellcheck disable=SC2154,SC2034,SC1090
 
-TMPDIR=${PT_tmpdir:-$(dirname "$PT_tarball")}
-PEDIR=$(dirname "$(tar -tf "$PT_tarball" | head -n 1)")
+declare PT__installdir
+source "$PT__installdir/deploy_pe/files/common.sh"
+PUPPET_BIN=/opt/puppetlabs/bin
 
+(( EUID == 0 )) || fail "This utility must be run as root"
 
-function raise_error {
-  cat << ERROR_MESSAGE
-  { "_error": {
-    "msg": "Task exited 1:\\n $1",
-    "kind": "deploy_pe/install_pe-error",
-    "details": { "exitcode": 1 }
-    }
-  }
-ERROR_MESSAGE
-  exit 1
-}
-
-function output {
-  cat << OUTPUT_MESSAGE
-  { 
-    "msg": "$1"
-  }
-OUTPUT_MESSAGE
-  exit 0
-}
-
-function validate_input {
-  if [ ! -e "$PT_tarball" ]; then
-    raise_error "Tarball, '$PT_tarball', is missing; exiting"
-  fi
-
-  if [ ! -e "$PT_pe_conf" ]; then
-    raise_error "pe.conf, '$PT_pe_conf', is missing; exiting"
-  fi
-
-  if [ ! -d "$TMPDIR" ]; then
-    mkdir -p "$TMPDIR"
-    if [ ! -d "$TMPDIR" ]; then
-      raise_error "Tmpdir, '$TMPDIR', does not exist and could not be created; exiting"
-    fi
-  fi
+# Hackish, but cd to the directory we get via the PT_tarball parameter
+# Extract it, cd to the directory with the .tar.gz suffix, and run the installer
+cd "${tarball%/*}" || fail
+tar xf "$tarball" || fail "Error extracting PE tarball"
```

Comment:
What versions of tar are required for the implicit `z` extraction? Do we have to worry about not having the right version on RHEL 6?

Reply:
I can test it on older an older OS.  The error check should catch this and present the `stderr` to the user.  I'm tempting to just let the resolution be "get yourself a real version of tar."

Reply:
It looks like centos 6 ships with a version of tar that supports this.